### PR TITLE
Don't keep a second copy of the model in the performance path

### LINKED ISSUE #4928: [release/2.14] huggingface DebertaV2ForQuestionAnswering inference amp_fp16 eager performance regression (~100x)
### 🐛 Describe the bug

Issue found during pre RC test with torch 2.14.0.dev20260805+xpu. 

There is performance regression on Windows with Intel XPU (Arc B580). Batch size used: 2

Testcase | Model | Inductor Perf status | Inductor Perf diff | Inductor PT2.13 latency [ms] | Inductor PT2.14 latency [ms] | Eager Perf status | Eager Perf diff | Eager PT2.13 latency [ms] | Eager PT2.14 latency [ms]
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
huggingface_performance_inference_amp-fp16 | DebertaV2ForQuestionAnswering | PASS | -2.70% | 44.57 | 43.36 | FAIL | 10898.30% | 70.84 | 7791.14

### Versions

<details>
<summary>Click to expand traceback</summary>
PyTorch version: 2.14.0.dev20260805+xpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro (10.0.26200 64-bit)
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: N/A

Python version: 3.12.13 | packaged by conda-forge | (main, Mar  5 2026, 16:36:12) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26200-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260100
Intel GPU driver version:
* 32.0.101.8864 (20260717000000.******+***)
Intel GPU models onboard:
* Intel(R) Arc(TM) B580 Graphics
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Arc(TM) B580 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE20B, uuid=86800be2-0000-0000-0300-000000000000, driver_version='1.15.38308', total_memory=11875MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=0MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Name: 13th Gen Intel(R) Core(TM) i5-13400
Manufacturer: GenuineIntel
Family: 205
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 2500
MaxClockSpeed: 2500
L2CacheSize: 9728
L2CacheSpeed: None
Revision: None

Versions of relevant libraries:
[pip3] dpcpp-cpp-rt==2026.1.0
[pip3] intel-cmplr-lib-rt==2026.1.0
[pip3] intel-cmplr-lib-ur==2026.1.0
[pip3] intel-cmplr-lic-rt==2026.1.0
[pip3] intel-opencl-rt==2026.1.0
[pip3] intel-openmp==2026.1.0
[pip3] intel-pti==1.0.1
[pip3] intel-sycl-rt==2026.1.0
[pip3] mkl==2026.1.0
[pip3] mkl-include==2024.2.0
[pip3] mkl-static==2024.2.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] onemkl-license==2026.1.0
[pip3] onemkl-sycl-blas==2026.1.0
[pip3] onemkl-sycl-dft==2026.1.0
[pip3] onemkl-sycl-lapack==2026.1.0
[pip3] onemkl-sycl-rng==2026.1.0
[pip3] onemkl-sycl-sparse==2026.1.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2023.1.0
[pip3] tbb-devel==2021.13.1
[pip3] tcmlib==1.5.0
[pip3] torch==2.14.0.dev20260805+xpu
[pip3] torchaudio==2.11.0.dev20260805+xpu
[pip3] torchvision==0.29.0.dev20260805+xpu
[pip3] triton-xpu==3.8.0+git1e2d42a0
[pip3] umf==1.1.0
[conda] dpcpp-cpp-rt                 2026.1.0                pypi_0              pypi
[conda] intel-cmplr-lib-rt           2026.1.0                pypi_0              pypi
[conda] intel-cmplr-lib-ur           2026.1.0                pypi_0              pypi
[conda] intel-cmplr-lic-rt           2026.1.0                pypi_0              pypi
[conda] intel-opencl-rt              2026.1.0                pypi_0              pypi
[conda] intel-openmp                 2026.1.0                pypi_0              pypi
[conda] intel-pti                    1.0.1                   pypi_0              pypi
[conda] intel-sycl-rt                2026.1.0                pypi_0              pypi
[conda] mkl                          2026.1.0                pypi_0              pypi
[conda] mkl-include                  2024.2.0                pypi_0              pypi
[conda] mkl-static                   2024.2.0                pypi_0              pypi
[conda] numpy                        1.26.4                  pypi_0              pypi
[conda] onemkl-license               2026.1.0                pypi_0              pypi
[conda] onemkl-sycl-blas             2026.1.0                pypi_0              pypi
[conda] onemkl-sycl-dft              2026.1.0                pypi_0              pypi
[conda] onemkl-sycl-lapack           2026.1.0                pypi_0              pypi
[conda] onemkl-sycl-rng              2026.1.0                pypi_0              pypi
[conda] onemkl-sycl-sparse           2026.1.0                pypi_0              pypi
[conda] optree                       0.13.0                  pypi_0              pypi
[conda] tbb                          2023.1.0                pypi_0              pypi
[conda] tbb-devel                    2021.13.1               pypi_0              pypi
[conda] tcmlib                       1.5.0                   pypi_0              pypi
[conda] torch                        2.14.0.dev20260805+xpu  pypi_0              pypi
[conda] torchaudio                   2.11.0.dev20260805+xpu  pypi_0              pypi
[conda] torchvision                  0.29.0.dev20260805+xpu  pypi_0              pypi
[conda] triton-xpu                   3.8.0+git1e2d42a0       pypi_0              pypi
[conda] umf                          1.1.0                   pypi_0              pypi

</details>

### PR DESCRIPTION
**Pull request by PatrykWilczewski**
Don't keep a second copy of the model in the performance path

Problem: https://github.com/intel/torch-xpu-ops/issues/4928 - huggingface DebertaV2ForQuestionAnswering, inference, amp fp16, batch size 2 regressed by +10898% on the eager iteration on an Arc B580 (11876 MB), while inductor stayed flat. More generally, every performance run of the dynamo benchmarks holds two full copies of the model resident.

Root cause: run_performance_test() and run_performance_test_non_alternate() begin with 'model = self.deepcopy_and_maybe_parallelize(model)', which rebinds a local while run_one_model() and run()'s model loop still hold the original, so both copies stay live for the whole measurement - 2 x 3347MB for this model.
That leaves the caching allocator no headroom: it reservers ~12GB on a 11876MB card, and Windows satisfies allocations beyond device capacity from host memory instead of failing, so buffers that land there are read over the host bus at ~4-7GB/s instead of ~370GB/s. The copy serves no purpose on that path: deepcopy_and_maybe_parallelize() only deepcopies unless --ddp/--fsdp is given; it cannot isolate the caller, because maybe_cast() has already covered the module in place before the copy is taken.

Fix: take the copy only when there is something to wrap

Validation: Arc B580, Windows 11, pytorch https://github.com/pytorch/pytorch/commit/d864943a877d29ac551ac02bab5db080a551bb7f; python benchmarks/dynamo/huggingface.py --performance --amp -d xpu -n10 --amp-dtype float16 --inference --backend=<eager/inductor> --cold-start-latency --disable-cudagraphs --batch size 2 --only <model>:
speedup ratio - eager / inductor
* DebertaV2ForQuestionAnswering: speedup 33.18 -> 1.48 (ratio collapses by 22x). Allocated 6695 -> 3347MB, reserved 11956 -> 8510 MB, device free 2 -> 2849 MB, with per-iteration traffic unchanged (1088 allocations, 7.06 GB)
* other models are unaffected: BertForMaskedLM 1.46, DistilBertForMaskedLM 1.37, GoogleFnet 1.41
* --accuracy unaffected: DebertaV2ForQuestionAnswering pass, BertForMaskedLM pass
* --training works, where the optimizer attaches to the model DistilBertForMaskedLM speedup ratio 1.28
* the second call site was exercised via 'python -m benchmarks.dynamo.huggingface --optimus vertical_opt ... --only DistilBertForMaskedLM', which routes to run_performance_test_non_alternate(): exit 0, speedup 0.93

Fixes: https://github.com/intel/torch-xpu-ops/issues/4928

### CODE REVIEW

**Review by kdrozd-dev — APPROVED** [review]

**Comment by PatrykWilczewski** [comment]
Not needed anymore, fixed by https://github.com/pytorch/pytorch/pull/196178
